spec/models/project_spec.rb

```diff
@@ -293,17 +293,39 @@
   end
 
   describe '.goal_between' do
-    let(:start_at) { 100 }
-    let(:ends_at) { 200 }
-    subject { Project.goal_between(start_at, ends_at).order(:id) }
+    let(:start_at)                 { 200 }
+    let(:ends_at)                  { 400 }
+    let!(:all_projects)            { projects_under_start_at + projects_over_ends_at + projects_goal_between }
+    let!(:projects_under_start_at) { create_list(:project, 3, goal: 199) }
+    let!(:projects_over_ends_at)   { create_list(:project, 3, goal: 401) }
+    let!(:projects_goal_between)   {
+      [ create(:project, goal: 200), create(:project, goal: 399), create(:project, goal: 400) ]
+    }
 
-    before do
-      @project_01 = create(:project, goal: 100)
-      @project_02 = create(:project, goal: 200)
-      @project_03 = create(:project, created_at: 300)
+    subject { Project.goal_between(start_at, ends_at) }
+
+    context "when there are both start_at and ends_at" do
```

Comment:
I much prefer to not use implicit subject, but I understand if it's the project standards.

```ruby
context 'when filtering by goal range' do
  it 'returns projects which goal stands between that range' do
    projects = Project.goal_between(200, 400)
    
    expect(projects).to match_array([with_goal_200, with_goal_300, with_goal_400])
  end
end
```